Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs

```diff
@@ -5223,7 +5177,8 @@ protected override void UpdateTractiveForce(float elapsedClockSeconds, float t,
                     float connectRodInertiaForcelbf = inertiaSpeedCorrectionFactor * connectRodInertiaAngleFactor * ConnectingRodWeightLb;
 
                     // Account for the position of the crosshead position. In other words it depends upon whether the Rods and Reciporating gear is above or below the axle.
-                    if (crankAngleRad > 0 && crankAngleRad < Math.PI)
+                    // The crosshead will be -ve for normalised angles between 0 - 180, and +ve for normalised angles between 180 - 360
+                    if (normalisedCrankAngleRad > 0 && normalisedCrankAngleRad < Math.PI )
```

Comment:
I still don't know why normalisedCrankAngleRad is used, when testing against crankAngleRad > 0 has the same effect

Reply:
crankAngleRad operates in the range 0-180-0 and therefore it will always be positive (ie it never goes negative, and hence the conditions in the if statement will always be met) in every cycle, and within every cycle there are two points that are the same. Hence it is not possible to differentiate between the up or down parts of the cycle.

normalisedCranAngleRad operates in the range 0-360 so it is possible to determine the up and down part of the cycle.

I have spent some time trying to get the best solution, and testing it to make sure that it works, so I would like to leave it as it is.

Reply:
I'm sorry, but I cannot approve it yet as it is really confusing. You say:
> crankAngleRad operates in the range 0-180-0 and therefore it will always be positive

but that's not true. In the code, crankAngleRad operates between -180º->0º->180º, so it can be positive and negative. I don't see a need to fix that, I'm OK with it. If that is working, then there is no need to use normalisedCrankAngle too in UpdateTractiveForce() (you can use it in UpdateFX if you wish). Condition 0<normalisedCrankAngle<180, is equivalent to 0<crankAngleRad